src/NuGet.Core/NuGet.ProjectModel/ProjectLockFile/NuGetLockFileUtilities.cs

```diff
@@ -0,0 +1,138 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using NuGet.Common;
+using NuGet.LibraryModel;
+using NuGet.Packaging.Core;
+using NuGet.Shared;
+
+namespace NuGet.ProjectModel
+{
+    public static class NuGetLockFileUtilities
+    {
+        public static bool IsNuGetLockFileSupported(PackageSpec project)
+        {
+            var restorePackagesWithLockFile = project.RestoreMetadata?.RestorePackagesWithLockFile;
+            return MSBuildStringUtility.IsTrue(restorePackagesWithLockFile) || File.Exists(GetNuGetLockFilePath(project));
+        }
+
+        public static string GetNuGetLockFilePath(PackageSpec project)
+        {
+            if (project.RestoreMetadata == null || project.BaseDirectory == null)
+            {
+                // RestoreMetadata or project BaseDirectory is not set which means it's probably called through test.
+                return null;
+            }
+
+            var path = project.RestoreMetadata.NuGetLockFilePath;
+
+            if (string.IsNullOrEmpty(path))
+            {
+                path = Path.Combine(project.BaseDirectory, "packages." + project.RestoreMetadata.ProjectName.Replace(' ', '_') + ".lock.json");
+
+                if (!File.Exists(path))
```

Comment:
`File.Exists is case insensitive and it can't be confusing.` 
I didn't understand what you mean by this?

Also, why should we use `ProjectPath` when there is already `ProjectName`? Are there cases where ProjectPath is set but not ProjectName in restore metadata? if not, then I'll lean towards using ProjectName and may be fixed that at other places going forward...

Reply:
It's not about whether it's set or not, but ProjectName defaults to the AssemblyName and PackageId.
It can easily get confusing. 

The project file name is more appropriate in this case.
Can't recall exactly at this moment why we append the ext in the g.props and g.targets, probably @emgarten remembers.

Reply:
discussed offline, for netcore project, `ProjectName` could be set to AssemblyName or PackageId so better to use `ProjectPath`